modules/samples/watertank/component/api/src/main/java/org/homeautomation/watertank/api/DeviceTypeServiceImpl.java

```diff
@@ -0,0 +1,434 @@
+            response.status(Response.Status.OK);
+            response.type("application/zip");
+            response.header("Content-Disposition", "attachment; filename=\"" + zipFile.getFileName() + "\"");
+            Response resp = response.build();
+            zipFile.getZipFile().delete();
+            return resp;
+        } catch (IllegalArgumentException ex) {
+            return Response.status(400).entity(ex.getMessage()).build();//bad request
+        } catch (DeviceManagementException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (JWTClientException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (APIManagerException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (IOException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        } catch (UserStoreException ex) {
+            log.error(ex.getMessage(), ex);
+            return Response.status(500).entity(ex.getMessage()).build();
+        }
+    }
+
+    /**
+     * Register device into device management service
+     *
+     * @param deviceId unique identifier for given device type instance
+     * @param name     name for the device type instance
+     * @return check whether device is installed into cdmf
+     */
+    private boolean register(String deviceId, String name) {
+        try {
+            DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+            deviceIdentifier.setId(deviceId);
+            deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+            if (APIUtil.getDeviceManagementService().isEnrolled(deviceIdentifier)) {
+                return false;
+            }
+            Device device = new Device();
+            device.setDeviceIdentifier(deviceId);
+            EnrolmentInfo enrolmentInfo = new EnrolmentInfo();
+            enrolmentInfo.setDateOfEnrolment(new Date().getTime());
+            enrolmentInfo.setDateOfLastUpdate(new Date().getTime());
+            enrolmentInfo.setStatus(EnrolmentInfo.Status.ACTIVE);
+            enrolmentInfo.setOwnership(EnrolmentInfo.OwnerShip.BYOD);
+            device.setName(name);
+            device.setType(DeviceTypeConstants.DEVICE_TYPE);
+            enrolmentInfo.setOwner(APIUtil.getAuthenticatedUser());
+            device.setEnrolmentInfo(enrolmentInfo);
+            boolean added = APIUtil.getDeviceManagementService().enrollDevice(device);
+            if (added) {
+                APIUtil.registerApiAccessRoles(APIUtil.getAuthenticatedUser());
+            }
+            return added;
+        } catch (DeviceManagementException e) {
+            log.error(e.getMessage(), e);
```

Comment:
repeating the same error message (i.e. e.getErrorMessage()) here, which can be found in the stack trace as well, is not a good practice.